src/memex/models/document.py

```diff
@@ -393,10 +395,14 @@ def merge_documents(session, documents, updated=None):
             m.document = master
             m.updated = updated
 
-        session.delete(doc)
-
     try:
         session.flush()
+        session.query(Annotation) \
+            .filter(Annotation.document_id.in_(duplicate_ids)) \
```

Comment:
I can change the filter to use `Annotation.document.in_()`, but I think we still need the `duplicate_ids` because we need to change the way we delete the duplicate documents. We have a foreign-key constraint on `annotation.document_id`, so we can't delete a document (or have it accidentally flushed to the db) while the annotations haven't changed yet. Plus this way we only do one query for changing the `document_id` of all Annotations that belong to one of the duplicate documents.